bi_sql_editor/hooks.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015-2017 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import SUPERUSER_ID
+from openerp.api import Environment
+
+
+def uninstall_hook(cr, registry):
+    env = Environment(cr, SUPERUSER_ID, {})
+    recs = env['bi.sql.view'].search([])
+    for rec in recs:
+        rec.button_set_draft()
+        rec.unlink()
+
+    # delete dirty data that could cause problems
+    # while re-installing the module
+    # Drop materialized views
+    cr.execute("""
+        select relname
+        from pg_class
+        where relname like 'x_bi_sql_view%' and relkind='m'
+        """)
+    for r in cr.fetchall():
+        cr.execute("""
+               DROP MATERIALIZED VIEW %s
+            """ % r)
+
+    cr.execute("""
+        select relname
+        from pg_class
+        where relname like 'x_bi_sql_view%' and relkind='r'
+        """)
+    for r in cr.fetchall():
+        cr.execute("""
+               DROP TABLE %s
+            """ % r)
+    cr.execute("""
+        select table_name from INFORMATION_SCHEMA.views
+        where table_name like 'x_bi_sql%'""")
+
+    # Drop not materialized views
+    for v in cr.fetchall():
+        cr.execute("""
+               DROP VIEW %s
+            """ % v)
+
+    # Drop table if uninstalling went wrong
+    cr.execute("""
+        delete from ir_model_fields where model like 'bi.sql.view%';
```

Comment:
I think the same. This is supposed to be done by this line `rec.button_set_draft()` But for some reason it does not work properly. At the end it shows this error:

> UserError: (u"The field 'bi.sql.view.field.bi_sql_view_id' cannot be removed because the field 'bi.sql.view.bi_sql_view_field_ids' depends on it.", None)

That's why I put all those queries.